### ISSUE TITLE
[IPFS] Add a third button: Use desktop client

### ISSUE THREAD

**Issue by RubenKelevra** [brave/brave-browser#16901]
## Description 
Currently, there's a nice dialog for IPFS:

![Screenshot from 2021-07-12 17-03-24](https://user-images.githubusercontent.com/614929/125311532-c4d40a80-e333-11eb-8696-2bebea213ba9.png)

But this is clearly missing a button for "Connect to Desktop client".

I think there are some people who might want to run IPFS as a dedicated process, not in the browser - so it survives when you close the browser. This has also the advantage of having a system tray icon.

## Reproduces how often: 
Easily reproduced

## Brave version (brave://version info)

Brave | 1.26.74 Chromium: 91.0.4472.124 (Official Build) (64-bit)

**Comment by lidel** [comment]
Yes, some power users may have [ipfs-desktop](https://docs.ipfs.io/install/ipfs-desktop/) or  a [standalone go-ipfs](https://docs.ipfs.io/install/command-line/) installed.

For now, they can set it as "public gateway" in the Settings:

> ![2021-07-13--14-38-28](https://user-images.githubusercontent.com/157609/125453017-5941dac5-bab7-4e7a-a3dd-daddd8c11746.png)

AFAIK only downside is that it won't keep native URI in address bar, and show localhost:8080 URLs.

- Brave could keep native URI when localhost gateway is used, but ONLY when quick test confirms  `http://localhost:8080/ipfs/bafkqae2xmvwgg33nmuqhi3zajfiemuzahiwss`  returns a redirect to unique Origin at `http://bafkqae2xmvwgg33nmuqhi3zajfiemuzahiwss.ipfs.localhost:8080/`  (we can't allow native URIs without Origin isolation). 

As for the activation UX, I would be very careful about overcrowding the activation screen:
- IF we add any additional button, it should be displayed ONLY when we confirm the local node is running on the default port. 
  - Use case we want to cover is "User installed and is running IPFS Desktop before activating IPFS in Brave" and nothing more.
  - When "IPFS activation screen" is displayed,  do is a quick redirect test mentioned above and then see if payload returned from `http://bafkqae2xmvwgg33nmuqhi3zajfiemuzahiwss.ipfs.localhost:8080/` match the expected  `Welcome to IPFS :-)`  and display additional button  "Use existing local gateway"  at the very top. 
    - Open question: what happens when local gateway is shut down. IPFS in Brave will be broken. We need to improve UX of error page and suggest changing "Gateway" URL when it fails.
- IF user is running on custom port, then no button is displayed: it is up to the user to set up Gateway  URL via Settings.
  - I advise against adding any "port" inputs on activation screen (looks scary to newcomers, prone to errors, adds maintenance complexity)

**Comment by nathanl4** [comment]
FYI, confusingly you cannot set http://127.0.0.1:8080 (like if you copy and paste out of IPFS desktop) to be your IPFS gateway as Brave returns a "Invalid gateway address" error. However, http://localhost:8080 works fine. I'd very much like to see easy Brave option to just use my desktop IPFS client since this was very confusing.
